src/Temporalio/Bridge/Metric.cs

```diff
@@ -51,18 +51,44 @@ internal class MetricInteger : SafeHandle
         /// </summary>
         /// <param name="value">Value to record.</param>
         /// <param name="attributes">Attributes to set.</param>
-        public void Record(ulong value, MetricAttributes attributes)
+        public void RecordInteger(ulong value, MetricAttributes attributes)
```

Comment:
Rather than allowing all record kinds on one `Metric` type it seems like it'd be pretty easy to hand out specialized versions with only one record method each corresponding to the kind passed in when created